### PR TITLE
Adding NWCK to repository

### PR DESCRIPTION
**Pull request by mgperson**

### CODE REVIEW

File: NWCK/src/NWCK.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+    def load_all_edges(self):
+        next_token = self.get_next_token()
+        while next_token != ';':
+            if next_token == '(':
+                self.load_edges()
+                next_token = self.get_next_token()
+
+    def get_GUID(self):
+        self.GUID += 1
```
Comment (jameswarren) [inline NWCK/src/NWCK.py]: using the name `GUID` typically implies it's assigned upon creation and never modified.

File: NWCK/src/NWCK.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+#Note: Might go back and use rosalind utilities Newick tree for this...
+
+
+class NWCK:
+    def __init__(self, Tk, xk, yk):
+        self.nwck_tree = Tk
+        self.edges = {}
+        self.GUID = 0
+        self.i = 0
```
Comment (jameswarren) [inline NWCK/src/NWCK.py]: could certainly use a better name if it's publicly accessible

File: NWCK/src/NWCK.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            for adjacent_node in self.edges[node[0]]:
+                if adjacent_node not in visited:
+                    queue.insert(0,(adjacent_node,node[1]+1))
+        return -1
+
+
+    def is_special_character(self,char):
+        return char in '(),;'
+
+    def get_next_token(self):
```
Comment (jameswarren) [inline NWCK/src/NWCK.py]: this would make uncle bob a sad panda

File: NWCK/src/NWCK.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+                nodes_at_level.append(self.load_edges())
+            elif next_token not in '(,':
+                nodes_at_level.append(next_token)
+            next_token = self.get_next_token()
+        next_token = self.get_next_token()
+        if self.is_special_character(next_token):
+            node = self.get_GUID()
+            self.i -= 1
+        else:
+            node = next_token
```
Comment (jameswarren) [inline NWCK/src/NWCK.py]: semantically the ideas of edges and nodes are straightforward, but their representations are not clear to someone reading the code

File: NWCK/src/NWCK.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+        self.add_edges(nodes_at_level,node)
+        return node
+
+    def add_edges(self,nodes,root_node):
+        for node in nodes:
+            self.edges[node] = self.edges.get(node,[]) + [root_node]
+            self.edges[root_node] = self.edges.get(root_node, []) + [node]
+
+    def get_distance_between_nodes(self,node_a,node_b):
+        #BFS of edges
```
Comment (jameswarren) [inline NWCK/src/NWCK.py]: interesting idea - in some cases BFS could be very quick, but in others not so much - depends on the branching factor of the tree. I think it's fine in this case, but as an alternative, you can find the paths for each node to the root to find the most recent common ancestor (the other solution does something like this)

File: NWCK/src/NWCK.py (hunk 3 above)
Comment (mgperson) [reply NWCK/src/NWCK.py]: :(

File: NWCK/src/NWCK.py (hunk 1 above)
Comment (mgperson) [reply NWCK/src/NWCK.py]: I have changed GUID to current_internal_node_number to reflect what this number actually represents in the model of the tree

File: NWCK/src/NWCK.py (hunk 2 above)
Comment (mgperson) [reply NWCK/src/NWCK.py]: i has been changed to index_of_tree_string to represent index of the parsing process. 
GUID has been changed to current_internal_node_number

File: NWCK/src/NWCK.py (hunk 5 above)
Comment (mgperson) [reply NWCK/src/NWCK.py]: Cool, will implement that as well

File: NWCK/src/NWCK.py (hunk 3 above)
Comment (mgperson) [reply NWCK/src/NWCK.py]: Have changed the parameter char to "character" Can see why that's bad. Have also removed "character" from the function name. Is there something else uncle bob would dislike too???

**Pull request merged**